### PR TITLE
Optimize conversations update on Frontend

### LINKED ISSUE #9630: 🔥 Sever performance/CPU issue with at least firefox due to Talk page rendering
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Have 200+ conversations
2. Have the tab open and others help sending messages so chats sort up and down regularly

### Expected behaviour
🚤 Everything works fast and properly

### Actual behaviour
🔥  PC heats up, CPU is 100% and Firefox eventually becomes unresponsive

## Talk app

**Talk app version:** 17rc2 at least

### PR DESCRIPTION
**Pull request by ShGKme**
### ☑️ Resolves

* Maybe fix #9630
* Decrease re-rendering time for `LeftSidebar` on re-fetching conversations every 30 seconds when:
  - External signaling is using
  - This is not full re-fetching every 5 minutes

### 🖼️ Screenshots

No visual changes

### 🚧 Tasks

- [x] Add new action `addConversationIfChange` to add new conversation only if it was moidified according to `lastActivity` and `modifiedSince`
- [x] Add new action `postAddConversation` for common post-action of adding a conversation
- [x] Use new `addConversationIfChange` in `fetchConversations` action
- [x] Fix old tests
- [x] Add new test for this case

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed 

### Alternative and additional solutions
- In comparison, compare `lastActivity` and old `lastActivity`, not `modifiedSince`
- In addition, cache conversations list with KeepAlive to make returning from search fast
  - See: https://github.com/nextcloud/spreed/pull/9833
- Instead, decrease number of any state changes by full property comparison
  - Works with any signaling
  - Also works on 5-minute full re-fresh
  - Not domain-specific, a bit more code, but universal solution
  - See: https://github.com/nextcloud/spreed/pull/9811
- Instead/In addition use virtual scrolling
  - Increase `LeftSidebar` performance in general in any case, including memory usage, interactive reaction and etc.
  - Tricky in scrolling part, e2e testing and *perhaps* more risky
  - See: https://github.com/nextcloud/spreed/pull/9811

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]
🤞🏼

File: src/store/conversationsStore.js [hunk 1]
```diff
@@ -203,14 +203,42 @@ const mutations = {
...
+	 *
+	 * @param {object} context dispatch context
+	 * @param {object} payload mutation payload
+	 * @param {object} payload.conversation the conversation
+	 * @param {number|0} payload.modifiedSince timestamp of last state or 0 if unknown
+	 */
+	addConversationIfChanged(context, { conversation, modifiedSince }) {
+		if (conversation.lastActivity >= modifiedSince) {
+			context.commit('addConversation', conversation)
+		}
```
Comment (Antreesy) [inline src/store/conversationsStore.js:228]:
```suggestion
		} else {
		  // Smart update
		}
```

Don't we want to apply smartUpdate here to cover rest cases (full re-fetch at least)?

**Review by Antreesy — APPROVED** [review]
Looks good from the code perspective (but still lagging for me because of internal signalling)

File: src/store/conversationsStore.js (hunk 1 above)
Comment (ShGKme) [reply src/store/conversationsStore.js:228]: If we use the smartUpdate approach in else, then we do not need `if` branch, as `smartUpdate` covers all possible updates anyway

**Comment by ShGKme** [comment]
> but still lagging for me because of internal signalling

To test on internal you can add:
```js
this.roomListModifiedBefore = ~~(Date.now() / 1000)
```

after this `if`

https://github.com/nextcloud/spreed/blob/d1adff82227f0cb3fc7099fb3196d2d10941fcb5/src/components/LeftSidebar/LeftSidebar.vue#L493-L499

**Pull request merged**